opennms-services/src/test/java/org/opennms/netmgt/vacuumd/AutomationProcessorTest.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2003-2021 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 2021 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.vacuumd;
+
+import junit.framework.TestCase;
+import org.junit.Assert;
+import org.opennms.netmgt.config.vacuumd.Action;
+import org.opennms.netmgt.config.vacuumd.Statement;
+
+public class AutomationProcessorTest extends TestCase {
+
+    public void testGetActionSQL() {
+        String emptyString = "";
+        //test null Action
+        AutomationProcessor.ActionProcessor ap1 = new AutomationProcessor.ActionProcessor("", null);
+        Assert.assertEquals(ap1.getActionSQL(), emptyString);
```

Comment:
The first parameter to `assertEquals` is the expected value, and the second is the actual value - so they should be switched.

Reply:
Good eye. I'll switch it and have another pull request to let you review.

Reply:
By the way, I wonder if we have SonarQube so something similar so that we can clear these warnings in Intellij before we commit code instead of manually find them?

Reply:
We actually do have Sonar running on the `develop` branch: https://sonarcloud.io/project/issues?id=OpenNMS_opennms&branch=develop

Reply:
That is great to know. Thank you.  Is there any doc I can refer to add sonar server to my Intellij.

Reply:
Thanks. I think I've figured out configuring the OpenNMS sonar server in my IntelliJ.